data.py

```diff
@@ -33,14 +33,19 @@ def thread_main(self, sess):
         while not stop:
             iterator = load_data()
             for data in iterator:
+                while self.queue_size.eval(session=sess) == self.max_queue_size:
+                    if self.coord.should_stop():
+                        break
+                    time.sleep(0.02)
```

Comment:
0.02 is just a suitable time I think. Don't have to be 0.02
without time.sleep. this loop will be expensive and might affect the performance. also sleep will make the program to have chance to switch back to main thread.  GIL is a problem.